request.go

```diff
@@ -589,6 +593,15 @@ func unmarshalAttribute(
 	value = reflect.ValueOf(attribute)
 	fieldType := structField.Type
 
+	i := reflect.TypeOf((*AttributeUnmarshaler)(nil)).Elem()
```

Comment:
Can this be a package level var so it doesn't have to be evaluated on every attribute?